compiler/qsc_doc_gen/src/display.rs

```diff
@@ -589,6 +692,38 @@ fn eval_functor_expr(expr: &ast::FunctorExpr) -> ty::FunctorSetValue {
     }
 }
 
+fn as_struct(ty_def: &ast::TyDef) -> Option<Vec<ast::FieldDef>> {
+    match ty_def.kind.as_ref() {
+        ast::TyDefKind::Paren(inner) => as_struct(inner),
```

Comment:
There're some recursion calls to remove parens here. We don't expect enormously large recursion here, do we?

Reply:
It's only as large as the user's code is silly 😊.
```qsharp
newtype Foo = ((((First: Int))), Second: Int);
```
This example will recurse over 3+1 calls to this function.